### ISSUE TITLE
nextcloud 22 new install App "Preview Generator" cannot be installed because it is not compatible with this version of the server.

### ISSUE THREAD

**Issue by asm301** [nextcloud/previewgenerator#243]
App "Preview Generator" cannot be installed because it is not compatible with this version of the server.

**Comment by PVince81** [comment]
@rullzer can you release it ?

**Comment by rullzer** [comment]
Not any time soon. I'm pretty busy and soon on holidays

**Comment by reiner-text** [comment]
Duplicate of #240

**Comment by ckujau** [comment]
Until this is fixed, the following appears to work just fine:
```
$ sudo sed 's/max-version="21"/max-version="22"/' -i.bak ../nextcloud/apps/previewgenerator/appinfo/info.xml
```

**Comment by beemeeup** [comment]
Is just did that and the message in the update says there are still updates missing for preview generator

Is there something else that needs to be run

**Comment by TS-CUBED** [comment]
It seems to work for me by simply installing it as unsupported (based on the info that the codebase is prepared for 22 and it's simply a max-version issue).

**Comment by asm301** [comment]
I thought it was fixed, but I still get App "Preview Generator" cannot be installed because it is not compatible with this version of the server.
If it is installed with the following command it will work
sudo -u www-data php occ app:enable previewgenerator --force

Ckujau solution works, but the integrity check fails so that is a separate thing totally just thought it should be mentioned

**Comment by cravas** [comment]
I'm running NC on docker and the owner of the files/folders is "abc" instead of "www-data"
When I bash into the container, I'm logged as root.

Can I edit the info.xml (to change the "max-version") as root, or do I need to do it as abc???

After editing it, will it assume the changes with a restart of the container or do I need to reinstall the app?

thanks for any info

**Comment by xkill** [comment]
@rullzer can you release it? People is opening duplicated bugs: #250 #240, and lots of people is waiting for this release to upgrade Nextcloud to 22

Thanks in advance!!!

**Comment by rullzer** [comment]
I don't maintain this app anymore.

**Comment by markuman** [comment]
the last PR that was merged did @MorrisJobke, maybe he can help.

**Comment by MorrisJobke** [comment]
I'm also not anymore with the project - sorry.

**Comment by markuman** [comment]
Does someone else from @nextcloud feels responsible for this project? Because it's in the nextcloud/ group...

**Comment by markuman** [comment]
@kesselb are you able to help?

**Comment by bcutter** [comment]
Doesn´t look like. Great if you want to update to a new major release of NC but you´re stuck because of that one last important app... which seems to be unmaintained furthermore. 😢

**Comment by flintstryker** [comment]
I had no problems when I disregarded the warning and installed anyway on version 22.

**Comment by bcutter** [comment]
> I had no problems when I disregarded the warning and installed anyway on version 22.

Of course you can just update NC. But did you check preview generator is actually working after the update?

**Comment by gm0110** [comment]
> > I had no problems when I disregarded the warning and installed anyway on version 22.
> 
> Of course you can just update NC. But did you check preview generator is actually working after the update?

I did that also and ignored the message and it seems that it is Not working. Planning on downgrading to 21 because of this. I have alot of photo’s and i get annoyed by loading them everytime what this app was designed to “fix”

**Comment by palto42** [comment]
I have upgraded to NC 22 and to me it seems that the preview generator basically works, but I had to re-generate the preview images with `occ preview:generate-all` like after new install of the app.

**Comment by nssatlantis** [comment]
@rullzer @MorrisJobke 

Sorry firstly for dragging any one back in, I understand you have stopped supporting this app and wish to move on.

However, could you please consider just smashing this into a release so that we can at the least, use this great app (easily) one more version?

I would be greatly appreciative if you could, and I believe a lot of users share that feeling.

If not, thanks for making this app in the first place and have a good day forward :)

**Comment by MorrisJobke** [comment]
That’s not possible for both of us as we both don’t have the keys to sign the release anymore.

**Comment by nssatlantis** [comment]
For those that have the capability of doing manual installations, I have quickly forked the latest and pushed it as a release, including the latest update.

https://github.com/nssatlantis/previewgenerator/releases/tag/v3.1.2

Comes as-is, 0 support, no credits to me in any way shape or form as I literally just hit 'create release' in the hopes this will work. 
If it doesn't work, I will remove the fork and that'll be that.

Also not tested for functionality, which I will be doing in the next few hours as a flu is grasping on for dear life.

**Comment by bcutter** [comment]
> This issue is simple to fix as explained earlier. I love this app it makes previews just as it is supposed to but if no one is maintaining it anymore then that is that  problem solved remove the preview generator

Can you please link that simple fix please? So everyone wondering why this issue has been closed can fix it on their own. Thanks.

**Comment by bcutter** [comment]
That works for almost every app, right. BUT does it actually work with new major version? There are reports stating previews aren't generated on NC v22.

**Comment by palto42** [comment]
My impression is that the preview generator doesn't properly work in NC 22 release. The "generate-all" initially seems to work, but I have some doubt with the "pre-generate" cron task.

**Comment by bcutter** [comment]
> 
> 
> My impression is that the preview generator doesn't properly work in NC 22 release. The "generate-all" initially seems to work, but I have some doubt with the "pre-generate" cron task.

Could you check this? Should be easy to test if previews are actually generated.
1. Upload new images
2. Wait for preview generator cron or run it manually
3. Check if new content is created in /nextcloud-data-directory/appdata_ocXXXXXXXXXX/preview (check timestamp)

**Comment by cravas** [comment]
> > My impression is that the preview generator doesn't properly work in NC 22 release. The "generate-all" initially seems to work, but I have some doubt with the "pre-generate" cron task.
> 
> Could you check this? Should be easy to test if previews are actually generated.
> 
> 1. Upload new images
> 2. Wait for preview generator cron or run it manually
> 3. Check if new content is created in `/nextcloud-data-directory/appdata_ocXXXXXXXXXX/preview` (check timestamp)


Running NC on docker with 2 cronjobs 2x a day (files:scan-all && previewgenerator:pre-generate)
I've updated to NC22 sometime ago (having changed the appinfo.xml before) and can say that it's still working as it should:
`2021-10-15T22:43:47.848431858Z 2021-10-15T22:43:47+00:00 Generating previews for /xxxxx/files/Asus Phone/2021/10/IMG_20211015_150646.jpg
2021-10-15T22:43:48.327436990Z 2021-10-15T22:43:48+00:00 Generating previews for /xxxxx/files/Asus Phone/2021/10/IMG_20211015_150617.jpg
2021-10-15T22:43:48.798542607Z 2021-10-15T22:43:48+00:00 Generating previews for /xxxxx/files/Asus Phone/2021/10/IMG_20211015_145230.jpg
2021-10-15T22:43:49.303600783Z 2021-10-15T22:43:49+00:00 Generating previews for /xxxxx/files/Asus Phone/2021/10/IMG_20211015_145224.jpg
2021-10-15T22:43:49.764752422Z 2021-10-15T22:43:49+00:00 Generating previews for /xxxxx/files/Asus Phone/2021/10/IMG_20211015_144158.jpg
2021-10-15T22:43:50.290868709Z 2021-10-15T22:43:50+00:00 Generating previews for /xxxxx/files/Asus Phone/2021/10/IMG_20211015_143821.jpg
2021-10-16T09:19:01.759549744Z Starting scan for user 1 out of 3 (xxxxx)
2021-10-16T09:19:47.393797088Z Starting scan for user 2 out of 3 (xxxxx)
2021-10-16T09:20:01.667395790Z Starting scan for user 3 out of 3 (xxxxx)
2021-10-16T09:20:47.912043523Z +---------+-------+--------------+
2021-10-16T09:20:47.912269540Z | Folders | Files | Elapsed time |
2021-10-16T09:20:47.912386927Z +---------+-------+--------------+
2021-10-16T09:20:47.912502185Z | 725     | 40574 | 00:01:46     |
2021-10-16T09:20:47.912538518Z +---------+-------+--------------+
2021-10-16T21:19:01.650035912Z Starting scan for user 1 out of 3 (xxxxx)
2021-10-16T21:19:48.122458936Z Starting scan for user 2 out of 3 (xxxxx)
2021-10-16T21:20:02.798831742Z Starting scan for user 3 out of 3 (xxxxx)
2021-10-16T21:20:49.478582745Z +---------+-------+--------------+
2021-10-16T21:20:49.478955779Z | Folders | Files | Elapsed time |
2021-10-16T21:20:49.479017149Z +---------+-------+--------------+
2021-10-16T21:20:49.479183369Z | 726     | 40576 | 00:01:48     |
2021-10-16T21:20:49.479230165Z +---------+-------+--------------+
2021-10-16T22:43:02.308476173Z 2021-10-16T22:43:02+00:00 Generating previews for /xxxxx/files/Asus Phone/2021/10/IMG_20211016_124726.jpg`

**Comment by daffydock** [comment]
Just to chip in. I already had it installed, and it did not work. Went in, and changed the Appinfo.xml to 22 --which you could do if you download, unpack, change the value, repack, and install. Ran it in terminal. No problems. It is currently running.

**Comment by sunjam** [comment]
Makes sense, but there just needs to be a better way to document which apps have gone unmaintained over time.  Any method that allows potential maintainers to get this sort of info would be really helpful!  Currently attempting to [track it on the forum](https://help.nextcloud.com/t/any-apps-not-yet-compatible-with-version-22/117204), but this process is super tedious when having to hand write and search for every app from version to version.  Perhaps a Github tag such as "Needs Maintainer" would prove helpful!
